pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java

```diff
@@ -172,7 +185,7 @@ public static URI getDeleteSegmentHttpUri(String host, int port, String rawTable
       String tableType)
       throws URISyntaxException {
     return new URI(StringUtil.join("/", StringUtils.chomp(HTTP + "://" + host + ":" + port, "/"), OLD_SEGMENT_PATH,
-        rawTableName + "/" + URIUtils.encode(segmentName) + TYPE_DELIMITER + tableType));
+        rawTableName + "/" + URIUtils.encode(segmentName) + "?" + TYPE_DELIMITER + tableType));
```

Comment:
This API looks to be deprecated. Why do we need this change?

Reply:
I'm changing `TYPE_DELIMITER` because I need to use `type=` instead of `?type=`. The current `getURI(String protocol, String host, int port, String path)` will quote `?`, I created another `getURI()` function to get the correct uri for start/end segment replacement api. Maybe we can merge theses two `getURI()` function.